WordPress/src/main/java/org/wordpress/android/widgets/PostListButtonType.kt

```diff
@@ -0,0 +1,29 @@
+package org.wordpress.android.widgets
+
+// PostListButton.java types - from attrs.xml
+enum class PostListButtonType constructor(val value: Int) {
+    BUTTON_NONE(0),
```

Comment:
Do we really need this button type? Just from a quick check, I couldn't find what role it's supposed to play in this `enum`. I don't know if you already had plans to do this refactor, but I am curious if we could get rid of the layout deciding which button type it's representing especially when it doesn't do anything about this information. Could we for example add the buttons to the layout without the type and update it programmatically? I believe we already do that in some cases, so streamlining that should make the code much cleaner. If we did that, I think we can get rid of `PostListButton` completely.

Reply:
The main reason why I kept the NONE type was to avoid having nullable buttonType in `PostListButton`. However, I can see it's not worth it.

>Could we for example add the buttons to the layout without the type and update it programmatically?

Yes, we can it's actually the purpose of the public `buttonType` setter. Question is whether we want to lose the ability to set the type from the XML. It's used for example in `post_cardview_skeleton` (it's also used in `PostViewHolder` but I'll remove this usage in the next related PR - #9431). 

If we get rid of the custom view, we'll need to set up the view on case by case basis. We could create a "helper" method which would take care of it, but I'm not sure what would be the best place to put this method to. We'd also need to create a ViewHolder for the loading state/skeleton. 

Imho this is right on the edge, but I'm leaning toward keeping the custom view. However, I'm totally fine removing it if you think it'd be better. I'd just like to remove it in the #9431 PR or just another PR after 9431 is merged so I don't have to update `PostViewHolder` and throw the changes away right after that.

I've made some changes in https://github.com/wordpress-mobile/WordPress-Android/pull/9430/commits/829b532b70cd2e8ae09f23f706aaa0d8e783fc3b

Reply:
I think we can change the skeleton and figure out the other minor issues, so I am leaning towards removing it, but I don't feel too strongly about it. I think you're in a better position to judge whether the effort is worth it for us, so whatever you decide I am onboard with it. Also, if you decide to remove it, we certainly don't have to do it in this PR.

I'll review the rest of the PR as is and merge it if it looks good. Thanks!

Reply:
👍 ok, I'll remove it after we merge #9431 ;).